sdks/agent-sdk/src/core/Agent.ts

```diff
@@ -192,7 +261,15 @@ export class Agent<ContentTypes> extends EventEmitter<
     }
   }
 
-  async #processMessage(message: DecodedMessage<ContentTypes>) {
+  async #processMessage(
+    message: DecodedMessage<ContentTypes>,
+    topic: keyof EventHandlerMap<ContentTypes> = "unhandledMessage",
+  ) {
+    // Skip messages with undefined content (failed to decode)
+    if (!hasDefinedContent(message)) {
+      return;
+    }
```

Comment:
not something to worry about in this PR, but in future we could potentially allow devs to capture these types of messages in the stream